core/modules/rproc/InfileMerger.cc

```diff
@@ -193,18 +159,43 @@ InfileMerger::InfileMerger(InfileMergerConfig const& c,
         return _deleteInvalidRows(jobAttempts);
     });
 
-    /* &&&INFILE
-    if (!_setupConnection()) {
-        throw InfileMergerError(util::ErrorCode::MYSQLCONNECT, "InfileMerger mysql connect failure.");
-    }
-    */
+
 }
 
 
 InfileMerger::~InfileMerger() {
 }
 
 
+void InfileMerger::_setEngineFromStr(std::string const& engineName) {
+    std::string eName;
+    for (auto&& c:engineName) { eName += toupper(c); }
+    if (eName == "INNODB") {
+        _dbEngine = INNODB;
+    } else if (eName == "MEMORY") {
+        _dbEngine = MEMORY;
+    } else if (eName == "MYISAM") {
+        _dbEngine = MYISAM;
+    } else {
+        LOGS(_log, LOG_LVL_ERROR, "unknown dbEngine=" << engineName << " using default MYISAM");
+        _dbEngine = MYISAM;
+    }
+    LOGS(_log, LOG_LVL_INFO, "set engine to " << engineToStr(_dbEngine));
+}
+
+std::string InfileMerger::engineToStr(InfileMerger::DbEngine engine) {
+    switch (engine) {
+        case MYISAM:
+            return "MYISAM";
+        case INNODB:
+            return "INNODB";
+        case MEMORY:
+            return "MEMORY";
+        default:
+            return "UNKNOWN";
```

Comment:
I would recommend throwing an exception here. Besides, the code would have to be rewritten like:
```
switch (engine) {
        case MYISAM: return "MYISAM";
        case INNODB: return "INNODB";
        case MEMORY: return "MEMORY";
}
// For others 
throw ....;
```
In this case the complier will also warn you if you missed explicitly mentioning enum values in the `switch`.